### ISSUE TITLE
AssemblyName.GetAssemblyName on zero byte file segfaults on all unix distros 

### ISSUE THREAD

**Issue by maryamariyan** [dotnet/coreclr#15544]
Running the following code snippet is expected to throw with BadImageFormatException but causes segmentation fault on every unix distribution

```
            try
            {
                string filename = Path.GetTempFileName();
                AssemblyName.GetAssemblyName(filename);
            }
            catch (BadImageFormatException)
            {
                throw;
            }
```

```
    frame #5: 0x00007f5d630f46d3 libcoreclr.so`PEDecoder::HasWriteableSections() const [inlined] PEDecoder::FindNTHeaders(this=0x00007f5cac00c928) const + 3 at pedecoder.inl:1211
    frame #6: 0x00007f5d630f46d0 libcoreclr.so`PEDecoder::HasWriteableSections(this=0x00007f5cac00c928) const at pedecoder.cpp:456
    frame #7: 0x00007f5d62d9bd66 libcoreclr.so`PEImage::GetLayoutInternal(unsigned int, unsigned int) [inlined] PEImage::CreateLayoutFlat(this=0x00007f5cac00c5d0) + 36 at peimage.cpp:1032
    frame #8: 0x00007f5d62d9bd42 libcoreclr.so`PEImage::GetLayoutInternal(this=0x00007f5cac00c5d0, imageLayoutMask=<unavailable>, flags=<unavailable>) + 98 at peimage.cpp:934
    frame #9: 0x00007f5d62d9a767 libcoreclr.so`PEImage::GetLayout(this=0x00007f5cac00c5d0, imageLayoutMask=<unavailable>, flags=<unavailable>) + 231 at peimage.cpp:875
    frame #10: 0x00007f5d62f550b2 libcoreclr.so`AssemblyNameNative::GetFileInformation(StringObject*) [inlined] PEImage::HasNTHeaders(this=0x00007f5cac00c5d0) + 33 at peimage.inl:204
    frame #11: 0x00007f5d62f55091 libcoreclr.so`AssemblyNameNative::GetFileInformation(filenameUNSAFE=<unavailable>) + 113 at assemblyname.cpp:60
```

in pedecoder.inl m_base seems to be null:
```
frame #5: 0x00007f5d630f46d3 libcoreclr.so`PEDecoder::HasWriteableSections() const [inlined] PEDecoder::FindNTHeaders(this=0x00007f5cac00c928) const + 3 at pedecoder.inl:1211
(lldb) frame variable
(const PEDecoder *) this = 0x00007f5cac00c928
(lldb) expr this->m_base
(TADDR) $0 = 0
```

**Comment by jkotas** [comment]
This is regression introduced by https://github.com/dotnet/coreclr/pull/10772

cc @ruben-ayrapetyan

**Comment by danmosemsft** [comment]
Improved title. Feel free to fix.

**Comment by ruben-ayrapetyan** [comment]
Hi @jkotas, @danmosemsft,

Sorry for taking so long to answer. I will take a look into this on the next week.

**Comment by ruben-ayrapetyan** [comment]
Fixed (#16747). @maryamariyan could you, please, verify this?

**Comment by maryamariyan** [comment]
@ruben-ayrapetyan have you seen the comments here? https://github.com/dotnet/corefx/pull/27737

**Comment by jkotas** [comment]
Fixed by #16747 / #16917
